config/configload/configload.go

```diff
@@ -0,0 +1,64 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// package configload implements the configuration Loader struct.
+package configload
+
+import (
+	"github.com/spf13/cast"
+	"github.com/spf13/viper"
+)
+
+const (
+	// ViperDelimiter is used as the default key delimiter in the default viper instance
+	ViperDelimiter = "::"
+)
+
+// Creates a new Viper instance with a different key-delimitor "::" instead of the
+// default ".". This way configs can have keys that contain ".".
+func NewViper() *viper.Viper {
+	return viper.NewWithOptions(viper.KeyDelimiter(ViperDelimiter))
+}
+
+// NewLoader creates a new Loader instanceg
+func NewLoader() *Loader {
+	return &Loader{
+		v: NewViper(),
+	}
+}
+
+// FromViper creates a Loader from a Viper instance
+func FromViper(v *viper.Viper) *Loader {
+	return &Loader{v: v}
+}
+
+// Loader loads configuration
+type Loader struct {
+	v *viper.Viper
+}
+
+// Viper returns the underlying Viper instance
+func (l *Loader) Viper() *viper.Viper {
+	return l.v
+}
+
+// UnmarshalExact unmarshals the config into a struct, erroring if a field is nonexistent
+func (l *Loader) UnmarshalExact(intoCfg interface{}) error {
+	return l.v.UnmarshalExact(intoCfg)
+}
+
+// ToStringMap creates a map[string]interface{} from a ConfigLoader
+func (l *Loader) ToStringMap() map[string]interface{} {
+	return cast.ToStringMap(l.v)
```

Comment:
Not sure this will return the expected result, we need to test this.

Reply:
I added some tests, looks like it doesn't so I brought back the implementation I had initially.